src/client/Microsoft.Identity.Client/Http/HttpManager.cs

```diff
@@ -189,7 +135,7 @@ private static HttpRequestMessage CreateRequestMessage(Uri endpoint, IDictionary
             }
 
             // package 500 errors in a "service not available" exception
-            if (IsRetryableStatusCode((int)response.StatusCode))
+            if ((int)response.StatusCode >= 500 && (int)response.StatusCode < 600)
             {
                 throw MsalServiceExceptionFactory.FromHttpResponse(
```

Comment:
Can you please add the status code in the error message and the uri that produced it? (just host and path, not full uri)

Reply:
Will be done as part of https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/4820/files. FYI @trwalke